src/Grid.js

```diff
@@ -57,26 +57,27 @@ var Grid = React.createClass({
       <div {...this.props} style={this.getStyle()} className="react-grid-Grid">
         <Header
           ref="header"
-          columns={this.state.columns}
-          onColumnResize={this.onColumnResize}
+          columnMetrics={this.props.columnMetrics}
+          onColumnResize={this.props.onColumnResize}
           height={this.props.rowHeight}
-          totalWidth={this.DOMMetrics.gridWidth()}
+          totalWidth={this.props.totalWidth}
           headerRows={headerRows}
           sortColumn={this.props.sortColumn}
           sortDirection={this.props.sortDirection}
+          onSort={this.props.onSort}
           />
         <div ref="viewPortContainer" onKeyDown={this.props.onViewportKeydown} onDoubleClick={this.props.onViewportDoubleClick}   onDragStart={this.props.onViewportDragStart} onDragEnd={this.props.onViewportDragEnd}>
             <Viewport
               ref="viewport"
-              width={this.state.columns.width}
+              width={this.props.columnMetrics.width}
```

Comment:
It comes from ColumnMetricsMixin in ReactDataGrid. I forgot to add it to the propTypes. Will fix

We should implement the AirBnB linting file with babel eslint so that we never miss a propType